src/util.h

```diff
@@ -53,6 +53,13 @@ namespace node {
 
 #define UNREACHABLE() ABORT()
 
+#define ASSIGN_OR_RETURN_UNWRAP(TYPE, PTR, OBJ, ...)                          \
+  do {                                                                        \
+    PTR = Unwrap<TYPE>(OBJ);                                                  \
```

Comment:
Alright, to fix this I had to do: https://github.com/nodejs/node/pull/6184/commits/f8d7dc8ed558b476069c262318c024193f8575dc

Both casts are necessary to get the correct pointer back.